search/errors.py

```diff
@@ -0,0 +1,41 @@
+from xml.dom import XML_NAMESPACE
+
+from flask import Blueprint, Response, abort, request
+
+from lxml import etree
+from lxml.builder import ElementMaker
+from werkzeug.exceptions import HTTPException
+
+NAMESPACE_MAP = {None: 'urn:ietf:rfc:7807'}
+
+
+def http_error_handler(error: HTTPException) -> Response:
+    doc = ElementMaker(nsmap=NAMESPACE_MAP)
+    # pylint: disable=no-member
+    error_doc = doc.problem(
+        doc.status(str(error.code)),
+        doc.title(error.name),
+        doc.detail(error.description),
+    )
+    # pylint: enable=no-member
+    error_doc.set('{%s}lang' % XML_NAMESPACE, "en")
+    response = etree.tostring(error_doc, xml_declaration=True, encoding='UTF-8')
+    return Response(response=response, status=error.code, mimetype='application/problem+xml')
+
+
+def get_error_blueprint() -> Blueprint:
+    blueprint = Blueprint('error', __name__)
+
+    @blueprint.route('/error', methods=['GET'])
+    def error() -> None:  # pylint: disable=unused-variable
```

Comment:
I remember this case before, I can't figure out which is the unused variable? Only `status_code` is assigned in this function, and it is used in `abort()`

Reply:
It's the function name. In this case `def error()` is the unused variable name. Pylint thinks a function declaration within another function that is then not called within the outer function is unused. But, Flask does map the inner function after being registered which will then be called when the route is accessed/requested.